service/history/shard/controller_test.go

```diff
@@ -427,6 +427,37 @@ func (s *controllerSuite) TestAcquireShardRenewLookupFailed() {
 	}
 }
 
+func (s *controllerSuite) TestAcquireShard_OwnershipLostErrorIsNotRetried() {
+	s.verifyAcquireShardAttempts(&persistence.ShardOwnershipLostError{}, 1, 1)
+}
+
+func (s *controllerSuite) TestAcquireShard_NonOwnershipLostErrorIsRetried() {
+	s.verifyAcquireShardAttempts(errors.New("temp error"), 2, 999)
+}
+
+func (s *controllerSuite) verifyAcquireShardAttempts(updateShardErr error, minAttempts int, maxAttempts int) {
+	s.T().Helper()
+	numShards := int32(1)
+	s.config.NumberOfShards = numShards
+	shardID := int32(1)
+	s.mockClusterMetadata.EXPECT().GetCurrentClusterName().Return(cluster.TestCurrentClusterName).AnyTimes()
+	s.mockClusterMetadata.EXPECT().GetAllClusterInfo().Return(cluster.TestSingleDCClusterInfo).AnyTimes()
+	s.mockServiceResolver.EXPECT().AddListener(gomock.Any(), gomock.Any()).Return(nil)
+	s.mockHostInfoProvider.EXPECT().HostInfo().Return(s.hostInfo).AnyTimes()
+	s.mockServiceResolver.EXPECT().Lookup("1").Return(s.hostInfo, nil).AnyTimes()
+	s.mockShardManager.EXPECT().GetOrCreateShard(gomock.Any(), getOrCreateShardRequestMatcher(shardID)).Return(
+		&persistence.GetOrCreateShardResponse{
+			ShardInfo: &persistencespb.ShardInfo{
+				ShardId: shardID,
+				Owner:   s.hostInfo.Identity(),
+			},
+		}, nil)
+	s.mockShardManager.EXPECT().UpdateShard(gomock.Any(), gomock.Any()).Return(updateShardErr).MinTimes(minAttempts).MaxTimes(maxAttempts)
+	s.mockServiceResolver.EXPECT().RemoveListener(gomock.Any())
+	s.shardController.Start()
```

Comment:
We should verify that it successfully acquired the shard when it is not ownership lost error

Reply:
Added `TestAcquireShardNonOwnershipLostErrorIsRetriedUntilSuccess`